addons/website/static/src/snippets/s_popup/options.js

```diff
@@ -92,16 +92,19 @@ options.registry.SnippetPopup = options.Class.extend({
     //--------------------------------------------------------------------------
 
     /**
-     * Moves the snippet in footer to be common to all pages
-     * or inside wrap to be on one page only
+     * Moves the snippet in #o_shared_blocks to be common to all pages or inside
+     * the first editable oe_structure in the main to be on current page only.
      *
      * @see this.selectClass for parameters
      */
     moveBlock: function (previewMode, widgetValue, params) {
-        const containerEl = this.$target[0].ownerDocument.querySelector(widgetValue === 'moveToFooter' ? 'footer' : 'main');
-        const whereEl = $(containerEl).find('.oe_structure:o_editable')[0];
+        const selector = widgetValue === 'allPages' ?
+            '#o_shared_blocks' : 'main .oe_structure:o_editable';
+        const whereEl = $(this.$target[0].ownerDocument).find(selector)[0];
```

Comment:
Good opportunity to stop using jQuery?

Reply:
JQuery is very useful here.. there is a custom CSS selector done with JQuery to select editable area.
```js
// jQuery extensions
$.extend($.expr[':'], {
    o_editable: function (node, i, m) {
        while (node) {
            if (node.className && _.isString(node.className)) {
                if (node.className.indexOf('o_not_editable') !== -1) {
                    return false;
                }
                if (node.className.indexOf('o_editable') !== -1) {
                    return true;
                }
            }
            node = node.parentNode;
        }
        return false;
    },
});
```
Do you want me to do a pure JS alternative ?

Reply:
No, you're right, it's fine. I missed it :+1: